homeassistant/components/mobile_app/entity.py

```diff
@@ -101,16 +102,19 @@ def device_info(self):
     @callback
     def _handle_update(self) -> None:
         """Handle async event updates."""
-        self._restore_pending_update()
+        self._apply_pending_update()
         self._async_update_attr_from_config()
         self.async_write_ha_state()
 
-    def _restore_pending_update(self) -> None:
+    def _apply_pending_update(self) -> None:
         """Restore any pending update for this entity."""
-        pending_updates = self.hass.data[DOMAIN].get(DATA_PENDING_UPDATES, {})
+        entity_type = self._config[ATTR_SENSOR_TYPE]
+        pending_updates = self.hass.data[DOMAIN][entity_type].get(
+            DATA_PENDING_UPDATES, {}
+        )
         if self._attr_unique_id in pending_updates:
```

Comment:
I'd use `dict.pop` here with a default so we don't need to access the dict more than once and so that the dict is updated, deleting the item here too, instead of having to do that later.


```suggestion
        if update := pending_updates.pop(self._attr_unique_id, None):
```